File: passctl/cmds/get.py
from rich.console import Console
from rich.table import Table
from passctl.log import error, success
from passctl.cmds import Command

class Get(Command):
    def __init__(self) -> None:
        super().__init__(
            "get", 
            "Read an entry from the vault",
        )

    def run(self) -> None:
        if len(self.args)<1:
            error("No entry specified")
        
        entry = self.args[0]
        vault = self.stg.data

        if not entry in vault.keys():
            error("Entry not found")

        console = Console()
        table = Table()        
        
        table.add_column("Username")
        table.add_column("Password")
        table.add_row(vault[entry]["user"], vault[entry]["pass"])

        console.print(table)

File: cmds/get.go
package cmds

import (
	"os"

	"github.com/passctl/client/config"
	"github.com/passctl/client/lib"
)

func Get() bool {
  if len(os.Args) != 3 {
    lib.Error("Specify a name for the entry")
    return false
  }

	if config.Cfg.Token == "" {
    lib.Error("You should first run the %ssetup%s command!",
      lib.COLOR_BOLD, lib.COLOR_RESET)
    return false
  } 

  master := GetMaster()
  err := config.LoadEntries(master)
  if err != nil {
    lib.ErrorMsg("Failed to load entries", err)
    return false
  }

  en := config.GetEntry(os.Args[2])
  if en == nil {
    lib.Error("Entry not found") 
    return false
  }

  lib.InfoVar("Username", en.User)
  lib.InfoVar("Password", en.Pwd)
  return true
}
